File: app/src/main/java/net/veldor/flibustaloader/dialogs/DonationDialog.java
package net.veldor.flibustaloader.dialogs;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;

import androidx.appcompat.app.AlertDialog;

public class DonationDialog {

    public static class Builder {
        private final Context mContext;

        public Builder(Context context) {
            mContext = context;
        }

        public AlertDialog build() {
            String[] donateOptions = new String[]{"PayPal", "Yandex"};
            // покажу диалог с выбором способа доната
            androidx.appcompat.app.AlertDialog.Builder builder = new androidx.appcompat.app.AlertDialog.Builder(mContext);
            builder
                    .setItems(
                            donateOptions,
                            (dialog, which) -> {
                                if (which == 0) {
                                    Intent intent = new Intent(Intent.ACTION_VIEW);
                                    intent.setData(Uri.parse("https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=YUGUWUF99QYG4&source=url"));
                                    mContext.startActivity(intent);
                                } else {
                                    Intent intent = new Intent(Intent.ACTION_VIEW);
                                    intent.setData(Uri.parse("https://money.yandex.ru/to/41001269882689"));
                                    mContext.startActivity(intent);
                                }
                            }
                    )
                    .setTitle("Поддержка разработки");
            return builder.create();
        }
    }

}

File: app/src/main/java/net/veldor/flibustaloader/dialogs/DonationDialog.kt
package net.veldor.flibustaloader.dialogs

import android.content.*
import android.net.Uri
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import net.veldor.flibustaloader.App
import net.veldor.flibustaloader.R


class DonationDialog {
    class Builder(private val mContext: Context) {
        fun build(): AlertDialog {
            val donateOptions = arrayOf("BitCoin", "Юmoney", "Patreon")
            // покажу диалог с выбором способа доната
            val builder = AlertDialog.Builder(mContext)
            builder
                .setItems(
                    donateOptions
                ) { _: DialogInterface?, which: Int ->
                    if (which == 0) {
                        val walletNumber = "38Q6K5g4Bcz1YkCscXRjcuZpW4tQn4vuhU"
                        val clipboard =
                            App.instance.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
                        val clip = ClipData.newPlainText("Copied Text", walletNumber)
                        clipboard.primaryClip = clip
                        Toast.makeText(mContext, mContext.getString(R.string.wallet_copied_message), Toast.LENGTH_LONG).show()
                    } else if(which == 1) {
                        val intent = Intent(Intent.ACTION_VIEW)
                        intent.data = Uri.parse("https://money.yandex.ru/to/41001269882689")
                        mContext.startActivity(intent)
                    }else {
                        val intent = Intent(Intent.ACTION_VIEW)
                        intent.data = Uri.parse("https://www.patreon.com/somedevf33434")
                        mContext.startActivity(intent)
                    }
                }
                .setTitle("Поддержка разработки")
            return builder.create()
        }
    }
}